### PR TITLE
feat: user project registration

### LINKED ISSUE #441: Add ability to register project under a user
In [proxy](https://github.com/radicle-dev/radicle-upstream/blob/master/proxy/src/registry/transaction.rs#L97), we can only register projects under orgs at the moment, but the UI implies that it's possible to register one under the current user too.

### PR DESCRIPTION
**Pull request by NunoAlexandre**
Closes #441 

### Changes

- Fully integrates with the latest registry `upstream` revision
- Full-stack changes to support user project registration alongside the already working org project registration

### Open questions

~~1. I see that we duplicate some of the registry types, such as `Id` and `ProjectName`. I followed the example for `ProjectDomain`, but I wonder if there's another motive to do this besides being able to derive certain traits that are not derived in the Registry?~~ #464 

~~2. How is `interface ProjectRegistration` used? It seems just like a bag of fields right now.~~

~~3. We currently have a `struct Project` defined in `http > org`. I made the code make it right so to keep this setup but it seems to me that this struct can better be generalized to work for both `user` and `org`?~~ #465

~~4. How should `API.md` be updated? Manually or is auto-generated by running some command? I went manual for now but it looks like it should be auto-generated.~~ #466

### CODE REVIEW

File: proxy/src/http/project.rs [hunk 1]
```diff
@@ -246,6 +248,15 @@ mod handler {
     }
 }
 
+impl From<(DomainType, registry::Id)> for registry::ProjectDomain {
```
Comment (xla) [inline proxy/src/http/project.rs:251]: This should live in `registry.rs`.

File: proxy/src/http/project.rs [hunk 2]
```diff
@@ -464,12 +475,24 @@ impl ToDocumentedType for MetadataInput {
     }
 }
 
+/// The domains we support under which a project can live.
```
Comment (xla) [inline proxy/src/http/project.rs:478]: Should also go in `registry.rs`.

Comment (NunoAlexandre) [reply proxy/src/http/project.rs:478]: My reasoning to have it here is that the single purpose of this type it to support the json deserialization of the request, having no other application for the registry.

File: proxy/src/http/project.rs (hunk 1 above)
Comment (NunoAlexandre) [reply proxy/src/http/project.rs:251]: Same as comment below.

File: proxy/src/http/project.rs [hunk 3]
```diff
@@ -480,9 +503,15 @@ impl ToDocumentedType for RegisterInput {
...
+            document::enum_string(vec!["org".into(), "user".into()])
+                .description("The type of domain the project will be registered under")
+                .example("user"),
+        );
+        properties.insert(
+            "domainId".into(),
             document::string()
-                .description("ID of the Org the project will be registered under")
+                .description("ID of the domain the project will be registered under")
                 .example("monadic"),
```
[the remark this answers is not in the document]
Comment (NunoAlexandre) [reply proxy/src/http/project.rs:515]: You are right! Will adjust :+1:

File: proxy/src/http/project.rs [hunk 4]
```diff
@@ -692,8 +723,56 @@ mod test {
...
+
+        let txs = cache.read().await.list_transactions(vec![])?;
+        let tx = txs.first().unwrap();
+
+        let have: Value = serde_json::from_slice(res.body()).unwrap();
+
+        assert_eq!(have, json!(tx));
+
+        Ok(())
+    }
```
[the remark this answers is not in the document]
Comment (NunoAlexandre) [reply proxy/src/http/project.rs:744]: @MeBrei could you expand on this suggestion? I don't quite follow. Would it mean to check that `tx` has one `message` that contains the fields we expect them to contain?

[the remark this answers is not in the document]
Comment (MeBrei) [reply proxy/src/http/project.rs:744]: I meant, to check that the transaction message actually includes the correct domain, ie the transaction not only exists but has the expected content

[the remark this answers is not in the document]
Comment (NunoAlexandre) [reply proxy/src/http/project.rs:761]: Got it (I think :smile:), thanks!

**Comment by xla** [comment]
> Why does that need to be addressed in this PR? This small change is already taking so long to get in, expanding the scope doesn't seem helpful at all.

Because we are introducing the functionality of registering projects under users and that entails exposing that semantically correct in the API. We can do it in a follow-up to keep velocity, but it's not a scope expansion.

**Comment by NunoAlexandre** [comment]
> > Why does that need to be addressed in this PR? This small change is already taking so long to get in, expanding the scope doesn't seem helpful at all.
> 
> Because we are introducing the functionality of registering projects under users and that entails exposing that semantically correct in the API. We can do it in a follow-up to keep velocity, but it's not a scope expansion.

Can you explain what is currently **wrong** with the current API? To me, it appears that this is about _improving_ something after criteria you haven't shared yet in concrete terms (better ergonomics), not a necessity, by any means, to have this feature achieved.

**Pull request merged**